src/dialogs/stat_dialogs/chart_options_view.pas

```diff
@@ -0,0 +1,286 @@
+unit chart_options_view;
+
+{$mode objfpc}{$H+}
+
+interface
+
+uses
+  Classes, SysUtils, stat_dialog_contribution, StdCtrls, ExtCtrls, EditBtn,
+  Controls, chart_options_model, executor, epidatafilestypes,
+  stat_dialog_custom_view;
+
+type
+
+  { TChartOptionsView }
+
+  TChartOptionsView = class(TCustomStatDialogView)
+  private
+    FDataModel: TChartOptionsModel;
+    FLabel: array of TLabel;
+    FText: array of TCustomEdit;
+    FDate: array of TDateEdit;
+    FOnModified: IStatDiaglogViewModified;
+    FSetXmin,
+    FSetXmax,
+    FSetYmin,
+    FSetYmax:  Boolean;
+  
+    procedure SetText(Sender: TObject);
+    procedure SetDate(Sender: TObject);
+//    procedure UpdateTexts();
+  public
+    constructor Create(TheOwner: TComponent); override;
+    procedure EnterView(); override;
+    function ExitView(): boolean; override;
+    function GetViewCaption(): UTF8String; override;
+    function IsDefined(): boolean; override;
+    procedure ResetView(); override;
+    procedure SetModel(DataModel: TChartOptionsModel);
+    procedure ResetMinMax;
+  end;
+
+implementation
+
+const
+  TITLE_TAG    = Ord(cbT);
+  FOOTNOTE_TAG = Ord(cbF);
+  XTITLE_TAG   = Ord(cbXT);
+  YTITLE_TAG   = Ord(cbYT);
+  COLOR_TAG    = Ord(cbC);
+  XMIN_TAG     = Ord(cbnXMin);
+  XMAX_TAG     = Ord(cbnXMax);
+  YMIN_TAG     = Ord(cbnYMin);
+  YMAX_TAG     = Ord(cbnYMax);
+
+  XDMIN_TAG    = Ord(cbdXMin);
+  XDMAX_TAG    = Ord(cbdXMax);
+  YDMIN_TAG    = Ord(cbdYMin);
+  YDMAX_TAG    = Ord(cbdYMax);
+
+{ TChartOptionsView }
+
+constructor TChartOptionsView.Create(TheOwner: TComponent);
+const
+  entryBox = 150;
+  dateSize = 120;
+var
+  EditText:  TCustomEdit;
+  DateText:  TDateEdit;
+  LabelText: TLabel;
+  PrevLabel: TLabel;
+  Labels:    array of UTF8String = ('Main Title', 'Footnote',
+             'X-Axis Title', 'Y-Axis Title', 'Colour Selection',
+             'X-Axis Minumum', 'X-Axis Maximum', 'Y-Axis Minumum', 'Y-Axis Maximum');
+  i:         Integer;
+begin
+  inherited Create(TheOwner);
+
+  SetLength(FText,  Ord(High(TChartOptionEdit)) + 1);
+  SetLength(FDate,  Ord(High(TChartMinMaxDate)) + 1);
+  SetLength(FLabel, Ord(High(TChartOptionEdit)) + 1);
+
+  // create labels
+
+  for i:= Low(FLabel) to High(FLabel) do
+    begin
+      LabelText := TLabel.Create(TheOwner);
+      with LabelText do
+        begin
+          Caption := Labels[i];
+          AnchorParallel(akLeft, 10, Self);
+          if (i = 0) then
+            AnchorParallel(akTop, 20, Self)
+          else
+            AnchorToNeighbour(akTop, 10, FLabel[i-1]);
+          Parent := self;
+        end;
+      FLabel[i] := LabelText;
+    end;
+
+  // create CustomEdit controls
+
+  for i := Low(FText) to High(FText) do
+    begin
+      EditText := TCustomEdit.Create(TheOwner);
+      with EditText do
+        begin
+          AnchorParallel(akTop, 0, FLabel[i]);
```

Comment:
No... that doesn't work because we have to be sure that date entry boxes are aligned to corresponding labels. To get this to work, I have to position the labels first and then anchor all input boxes to the appropriate label. I just could not get things to line up when I positioned the input boxes first. See next commit.